WalletWasabi/WabiSabi/Client/LinqEx.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections.Generic;
+
+namespace WalletWasabi.WabiSabi.Client;
+
+public static class LinqEx
+{
+	public static IEnumerable<T> TakeUntil<T>(this IEnumerable<T> list, Func<T, bool> predicate)
```

Comment:
Put this to `LinqExtensions` instead of creating a new file

Reply:
Done https://github.com/zkSNACKs/WalletWasabi/pull/8679/commits/12161774fbb26595d04099cb5c95a6bd6987a039